pulseguardian/guardian.py

```diff
@@ -219,21 +218,54 @@ def back_to_normal_email(self, user, queue_data):
            queue_data['messages'], self.del_queue_size)
 
         if self.emails and user.email is not None:
-            sendemail(subject=subject, from_addr=config.email_from,
-                      to_addrs=[user.email], username=config.email_account,
-                      password=config.email_password, text_data=body,
-                      server=config.email_smtp_server,
-                      port=config.email_smtp_port,
-                      use_ssl=config.email_ssl)
+            sendemail(subject=subject, to_addrs=[user.email], text_data=body)
+
+
+    def notify_connection_error(self):
+        if self.emails:
+            admin_emails = [user.email for user in User.query.filter_by(admin=True)]
+            subject = "Pulse error: Can't connect to Pulse"
+            body = '''
+                    Could not connect to Pulse.
+                    Details:
+                    Rabbit Url: {0}
+                    Rabbit User: {1}
+                '''.format(config.rabbit_management_url, config.rabbit_user)
+            sendemail(subject=subject, to_addrs=admin_emails, text_data=body)
+            logging.error(subject + body)
+
+
+    def notify_unknow_error(self, e):
+        if self.emails:
+            admin_emails = [user.email for user in User.query.filter_by(admin=True)]
+            subject = "Pulse error: Unknown error"
+            body = '''
+                    Unknown error occured.
+                    Details:
+                    {0}
+                '''.format(e)
+            sendemail(subject=subject, to_addrs=admin_emails, text_data=body)
+            logging.error(subject + body)
+
 
     def guard(self):
         logging.info("PulseGuardian started")
         while True:
-            queues = self.api.queues()
-            if queues:
-                self.monitor_queues(queues)
-            self.clear_deleted_queues(queues)
-            time.sleep(config.polling_interval)
+            try:
+                queues = self.api.queues()
+                if queues:
+                    self.monitor_queues(queues)
+                self.clear_deleted_queues(queues)
+                time.sleep(config.polling_interval)
+            except (TypeError, requests.ConnectionError):
+                #for some reason, when ConnectionError occurs on pulse, TypeError is raised
+                self.notify_connection_error()
```

Comment:
It would be best to pass the exception in, which probably has information about the error (e.g. unreachable host, closed port, etc.).